interface.go

```diff
@@ -2,7 +2,7 @@ package go_logger
 
 // Logger is logger for logger
 type Logger interface {
-	ErrorWithFields(message string, fields map[string]interface{})
+	ErrorWithFields(err error, fields map[string]interface{})
```

Comment:
This will be backward incompatible change. We will need to migrate the hell out of all projects.
I'm not saying it's not right, I think it should be the error here but this will be quite a migration task.

Reply:
Actually, this interface is used only internally. It exists so it can be mocked, if any project needs for testing purposes with `InitWithInstance` function.

Reply:
And this should be released with a major change version

Reply:
Roger that. Fair points!

Reply:
@jkrnak according to Aguiar this is not a breaking change.
So it will be released with a minor version.